src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -291,10 +291,22 @@ private async Task PublishFunctionApp(Site functionApp, GitIgnoreParser ignorePa
 
             Func<Task<Stream>> zipStreamFactory = () => ZipHelper.GetAppZipFile(workerRuntimeEnum, functionAppRoot, BuildNativeDeps, NoBuild, ignoreParser, AdditionalPackages, ignoreDotNetCheck: true);
 
+            var fileNameNoExtension = string.Format("{0}-{1}", DateTimeOffset.UtcNow.ToString("yyyyMMddHHmmss"), Guid.NewGuid());
+            if (functionApp.IsLinux && functionApp.IsDynamic)
+            {
+                if (workerRuntimeEnum == WorkerRuntime.python && !NoBuild && BuildNativeDeps)
```

Comment:
Just wondering, why do squashfs only for `--build-native-deps`? Is it because of the common dependency on docker?

Reply:
yes. At least for now since this is the scenario that needs it the most. however the only dependency we need is `./mksquashfs` which is available for linux, mac and Windows. So we can check if it's installed and use it to package instead of requiring docker. 

I'll need to refactor all of this action and all the helper classes to get that wired up correctly though.

Reply:
That makes sense. It's fair to leave it for `--build-native-deps` for now then. We can revisit later in needed.